db/db-design-notes.md

```diff
@@ -0,0 +1,193 @@
+# Database Design Notes
+
+-------------------------------------------------------------------------------
+## 2021-08-12 (Robyn)
+
+
+### Journeys table
+- I have removed the complex journey name column and just put in origin and 
+  destination columns. I don't think we should bother that it's technically not 
+  3NF as it makes our lives much easier. The origin and destination can just be
+  stored directly in the db when a user saves their journey.
+
+
+### Carbon saved / emissions table
+- Column added for emissions saved.
+- The plan is that a user will enter origin, destination and vehicle. The total 
+  emissions for the journey with the specified vehicle is returned alongside a 
+  list of alternative vehicle options and their associated emissions. The user 
+  tells the app if they want to save their journey and which vehicle they ended 
+  up using. It is the data of the journey made and vehicle selected as the one 
+  they used that gets stored in the DB. The app also calculates the difference 
+  between their originally chosen vehicle and the vehicle they end up using and 
+  **stores this in the emissions saved column** for that journey.
+
+**TO DO**: Consider how carbon saved will be entered into table. Will it also 
+be done in SQL or will it be through a Python function when the user opts to 
+save their journey. Whichever we decide, it is probably best that both the 
+carbon emitted and carbon saved functions are in the same place/language.
+
```

Comment:
I think, we should convert the emission calculation function into Python as well. 
We could potentially create a class that has methods that calculates emissions, offset etc. 
Would seem more standardized I guess ?